internal/commands/add_buildpack_registry_test.go

```diff
@@ -27,49 +28,62 @@ func TestAddBuildpackRegistry(t *testing.T) {
 func testAddBuildpackRegistryCommand(t *testing.T, when spec.G, it spec.S) {
 	when("AddBuildpackRegistry", func() {
 		var (
-			outBuf     bytes.Buffer
-			logger     = ilogging.NewLogWithWriters(&outBuf, &outBuf)
-			tmpDir     string
-			configFile string
-			assert     = h.NewAssertionManager(t)
+			outBuf  bytes.Buffer
+			logger  = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+			tmpDir  string
+			assert  = h.NewAssertionManager(t)
+			command *cobra.Command
 		)
 
 		it.Before(func() {
 			var err error
-			tmpDir, err = ioutil.TempDir("", "pack-home-*")
+			tmpDir, err = ioutil.TempDir("", "pack-home-")
 			assert.Nil(err)
 
-			configFile = filepath.Join(tmpDir, "config.toml")
+			assert.Nil(os.Setenv("PACK_HOME", tmpDir))
```

Comment:
Parallel test runs don't run in their own environment. I had it implemented this way but flakes started to occur due to the `PACK_HOME` being overridden in concurrent executions when `spec.Parallel()` is set. We could move it to `spec.Sequential` and live with it but ultimately it steered me into the thought that we _shouldn't_ be calling `DefaultConfigPath()` during our tests at all. Instead we should be relying on dependency injection instead of global state.

Reply:
Ah interesting. I thought this was an overthought, so I switched it back to our normal pattern. If we switch it to dependency injection, we should probably be doing it throughout though...

Reply:
We should. I didn't want to grow scope on someone else's PR. :D

Reply:
Makes sense. I'll revert this change, then; and I opened https://github.com/buildpacks/pack/issues/852 for fixing this entirely.